packages/headless/src/controllers/commerce/core/breadcrumb-manager/headless-core-breadcrumb-manager.ts

```diff
@@ -238,11 +238,13 @@ export function buildCoreBreadcrumbManager(
           .map((facetId) =>
             options.facetResponseSelector(engine[stateKey], facetId)
           )
-          .filter(
-            (facet): facet is AnyFacetResponse =>
-              facet !== undefined &&
-              facet.values.length > 0 &&
-              facet.values.some((value) => value.state !== 'idle')
+          .filter((facet): facet is AnyFacetResponse =>
+            facet !== undefined &&
+            facet.values.length > 0 &&
+            facet.type !== 'hierarchical'
+              ? facet.values.some((value) => value.state !== 'idle')
+              : findActiveValueAncestry((facet as CategoryFacetResponse).values)
+                  .length > 0
```

Comment:
This isn't easy to read.
You should extract the filter callback function into a const:
```ts
  const hasActiveValue = (facet): facet is AnyFacetResponse => {
    if (!facet) {
      return false;
    }
    if (facet.values.length === 0) {
      return false;
    }
    return facet.type === 'hierarchical'
      ? findActiveValueAncestry(facet.values).length > 0
      : facet.values.some((value) => value.state !== 'idle');
  };
```

It will make the array functions chain more readable.

Reply:
Second read: breadCrumbs should be the result of a single flatMap call.
We have an iteration galore here, and that's often a smell.

Reply:
Much more readable indeed! Thanks!

I applied your suggestion with some adjustments